File: php/Stack.php
<?php

class Stack {
    private $items = array();

    public function pop() {
        // return array_pop($this->items);
        $length = count($this->items);
        $popped = $this->items[$length - 1];
        $this->items = array_slice($this->items, 0, $length - 1);
        return $popped;
    }

    public function push($value) {
        // array_push($this->items, $value);
        $this->items[] = $value;
    }

    public function __toString() {
        $output = "TOP\n";
        for($i = count($this->items) - 1; $i >= 0; $i--){
            $output .= "{$this->items[$i]}\n";
        }
        $output .= "BOTTOM\n";
        return $output;
    }

    public function peek() {
        return $this->items[count($this->items) - 1];
    }
}

/* Test
$s = new Stack();
$s->push(1);
$s->push(2);
echo 'Peek ', $s->peek(), "\n";
$s->push(3);
$s->push(4);
echo 'Popped ', $s->pop(), "\n";
$s->push(5);
echo 'Popped ', $s->pop(), "\n";
echo 'Peek ', $s->peek(), "\n";
echo 'Popped ', $s->pop(), "\n";
$s->push(6);
echo 'Peek ', $s->peek(), "\n";
echo $s;
*/

File: python/Stack.py
class Stack(object):
    items = []

    def push(self, value):
        self.items.append(value)

    def pop(self):
        popped = self.items[len(self.items)-1]
        self.items = self.items[:-1]
        return popped

    def isempty(self):
        if not self.items:
            return True
        return False

    def peek(self):
        return self.items[-1]

    def __repr__(self):
        output = "TOP\n"
        i = len(self.items) - 1
        while i >= 0:
            output += str(self.items[i]) + "\n"
            i -= 1
        output += "BOTTOM"
        return output
        

## Test
# if __name__ == '__main__':
#     s = Stack()
#     s.push(1)
#     s.push(2)
#     print 'Peek %s' % s.peek()
#     s.push(3)
#     s.push(4)
#     print 'Popped %s' % s.pop()
#     s.push(5)
#     print 'Popped %s' % s.pop()
#     print 'Peek %s' % s.peek()
#     print 'Popped %s' % s.pop()
#     s.push(6)
#     print 'Peek %s' % s.peek()
#     print s
